astropy/logger.py

```diff
@@ -35,7 +35,7 @@
 ]
 for level in logging_levels:
     globals()[level] = getattr(logging, level)
-__all__ += logging_levels
+__all__ += logging_levels  # noqa: PLE0605
```

Comment:
I think the ``logging_levels`` variable can be created in conjunction with `__all__`, similar to this example where `a` -> `__all__` and `b` -> `logging_levels`. Then this whole line can be removed.

<img width="258" alt="CleanShot 2023-09-25 at 17 03 28@2x" src="https://github.com/astropy/astropy/assets/8949649/b1fb70fa-ad31-4cff-9ecd-0bb2a16f6087">

Reply:
@nstarman, Do you mean as follows?
```
__all__ += (logging_levels := ...
```
I tried and we still get ruff warning. Are you ok to keep the original change?

Reply:
Hm. I don't think we should be getting a `ruff` warning from that. As in, if we are there's a bug in ruff!

```python
__all__ =  ["a", "b"]
__all__ += (logging_levels := ["c", "d"])
```

Should be identical to 

```python
__all__ =  ["a", "b"]
__all__ += ["c", "d"]
```

from the perspective of ``__all__``.

Reply:
I opened https://github.com/astral-sh/ruff/issues/7672.
But it's fine to add the `noqa` for now.

Reply:
Thanks. In that case, the PR can be merged assuming there are no other concerns? I did make the other change requested already.

Reply:
https://github.com/astral-sh/ruff/issues/7672 is actually fixed, so if you can make the change to use the walrus operator and also add the noqa that would be great.
When we update our pre-commit it should remove the noqa!

Reply:
As in do
```python
__all__ += (logging_levels := ["c", "d"])  # noqa: PLE0605
```